### ISSUE TITLE
Feature Request: Connector Axis Controller additional methods

### ISSUE THREAD

**Issue by sam-snodgrass** [cwapi3d/cwapi3dpython#98]
This is a relatively large issue that highlights functionality that I would like to see in the API to match that of the controller axes in the application.

### Controller Axis section

1. Connector_axis_controller.get_standard_connector_list() -> list[int|str]
     - Return list of standard connector names, match implementation in C++

2. Connector_axis_controller.get_bolt_diameter(connector_id) -> float
     - Return diameter (float) of connector, match implementation in C++

3. Connector_axis_controller.get_catalog_connector_list -> list[int|str]
     - Return a list of catalog connector ids and names

4. Connector_axis_controller.import_from_file
     - Import from file, same implementation in C++

5. Connector_axis_controller.set_connector(connector_axis_id, catalog_connector_id) -> connector_id
     - Set a connector to the specific axis, same result as using this menu from the application, adds the selected connector to the axis at the start point.
     - Standard connector methods must function with catalog connectors as well. i.e. ca.set_bolt_length, ca.set_drilling_diameter, ca.set_section_drilling_diameter, etc…
<img width="357" alt="image" src="https://github.com/cwapi3d/cwapi3dpython/assets/145357460/c1206df0-732f-40c4-9dbe-1f771ea2320c">

### Counterbore section

6. Get_counter_bore_connector_list() -> list[int|str]
       - Return a list of available counterbore connector ids and names

7. Add_counterbore_connector_to_axis(counterbore_connector_id, connector_axis_id, start_point, end_point)
      - My suggestion for this is to select 2 points along the length of the connector axis and the counterbore_connector will generate between them, the way it works in the application is to select a border between zones and then you can add it there and modify it with diameter and depth settings.

8. set_depth(counterbore_connector_id, depth)

9. set_depth_automatic(counterbore_connector_id)

10. set_diameter(counterbore_connector_id, diameter)

11. set_diameter_automatic(counterbore_connector_id)

12. set_conic(counterbore_connector_id)

13. duplicate_to_end()
<img width="362" alt="image" src="https://github.com/cwapi3d/cwapi3dpython/assets/145357460/0cd8c785-2485-4b97-87b5-f8175202039d">

14. Connector_axis_controller.get_connectors_inside_axis(connector_axis_id)
       - Returns a list of all the connectors and counterbore_connectors inside of the axis

**Comment by kdurando** [comment]
> functionality that I would like to see in the API to match that of the controller axes in the application

@jspaquet I was discussing with our modeler today, and I just want to clarify that, at the best of our knowledge, number 7 above (And possibly 8 to 12) :

> 7. Add_counterbore_connector_to_axis(counterbore_connector_id, connector_axis_id, start_point, end_point)

> My suggestion for this is to select 2 points along the length of the connector axis and the counterbore_connector will generate between them, the way it works in the application is to select a border between zones and then you can add it there and modify it with diameter and depth settings.

 is not a feature yet possible with the app. So our need is not just a match of the app, as stated by Samuel, but rather extend to the implementation of the feature.

**Comment by kdurando** [comment]
@jspaquet I have a question. Has this issue/feature been implemented? This question only refer to matching the UI. I completely understand that the API can't provide features Cadwork does not allow with its UI.

**Comment by jspaquet** [comment]
@kdurando Yes, this has been implemented. We haven't merged the documentation changes yet but you can have a look at https://github.com/cwapi3d/cwapi3dpython/pull/222 if you want to have a look before we merge.

**Comment by kdurando** [comment]
Awesome! Thank you @jspaquet. We will give it a try and let you know.
